pkg/core/manager/workflow.go

```diff
@@ -42,6 +46,28 @@ func (mgr *workflowManager) Delete(ctx context.Context, name string) error {
 }
 
 func (mgr *workflowManager) AssignToCodeset(ctx context.Context, name, codesetProject, codesetName string) (wfListener *domain.WorkflowListener, webhookID *int64, err error) {
+	_, err = mgr.workflowStore.GetWorkflow(ctx, name)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	codeset, err := mgr.codesetStore.Find(ctx, codesetProject, codesetName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	wfListener, err = mgr.workflowBackend.CreateWorkflowListener(ctx, name, createWorkflowListenerTimeout*time.Minute)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	webhookID, err = mgr.codesetStore.CreateWebhook(ctx, codeset, wfListener.URL)
+	if err != nil {
+		return nil, nil, err
```

Comment:
What happens with the listener created earlier if this fails ?

Reply:
It is left there, should it be deleted? If that is the case the `CreateWorkflowListener` function needs to be split a bit and I propose to do that in a different PR.

Reply:
OK